mysql-test/suite/innodb/include/temp_table_encrypt.inc

```diff
@@ -181,3 +204,60 @@ INSERT INTO t1 VALUES (3);
 SET big_tables=1;
 SELECT * FROM t1 WHERE a IN(SELECT MAX(a) FROM t1);
 DROP TABLE t1;
+
+#
+# PS-5736: Make innodb_temp_tablespace_encrypt truly dynamic
+#
+
+--echo # innodb temp table encrypt is ON:
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TEMPORARY TABLE t1 (a INT);
+
+CREATE TEMPORARY TABLE tmp1 (a TEXT) ENCRYPTION='y';
+INSERT INTO tmp1 VALUES ('Maecenas condimentum leo eu lorem aliquam malesuada');
+CREATE TEMPORARY TABLE tmp2 (a TEXT) ENCRYPTION='y';
+
+--let $wait_condition= SELECT variable_value = 0 FROM performance_schema.global_status WHERE LOWER(variable_name) = 'innodb_buffer_pool_pages_dirty'
+--source include/wait_condition.inc
+
+--let $grep_pattern= Maecenas condimentum leo eu lorem aliquam malesuada
+--let $grep_file= $MYSQL_DATA_DIR/ibtmp1
+--let $grep_output= boolean
+--source include/grep_pattern.inc
+
+# file-per-table temporary tables do not obey 'innodb_temp_tablespace_encrypt'
+CREATE TEMPORARY TABLE t1 (a INT) KEY_BLOCK_SIZE=8 ENCRYPTION='y';
+DROP TABLE t1;
+CREATE TEMPORARY TABLE t1 (a INT) KEY_BLOCK_SIZE=8 ENCRYPTION='n';
+DROP TABLE t1;
+
+SET GLOBAL innodb_temp_tablespace_encrypt = OFF;
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TEMPORARY TABLE t1 (a INT) ENCRYPTION='y';
+
+INSERT INTO tmp2 VALUES ('Nam vestibulum mauris massa');
+
+CREATE TEMPORARY TABLE tmp3 (a TEXT);
+INSERT INTO tmp3 VALUES ('Sed sodales ligula sed enim condimentum');
+
+--let $wait_condition= SELECT variable_value = 0 FROM performance_schema.global_status WHERE LOWER(variable_name) = 'innodb_buffer_pool_pages_dirty'
+--source include/wait_condition.inc
+
+--let $grep_pattern= Nam vestibulum mauris massa
+--let $grep_file= $MYSQL_DATA_DIR/ibtmp1
+--let $grep_output= boolean
+--source include/grep_pattern.inc
```

Comment:
Please use include/search_pattern_in_file.inc instead of grep_pattern.inc.
search_pattern_in_file has cool feature "let ABORT_ON=FOUND;" or NOT_FOUND

With search_pattern_in_file.inc aborts instead of showing result content mismatch (with grep_pattern.inc). When the test aborts with all logs .ibd file, it much more useful in debugging.

Please fix all instances of grep_pattern.inc

Reply:
OK. Will use search_pattern_in_file instead